RecoLocalCalo/HcalRecAlgos/interface/HcalDeterministicFit.h

```diff
@@ -124,22 +120,22 @@ void HcalDeterministicFit::apply(const CaloSamples & cs, const std::vector<int>
   float n5=0;
   getLandauFrac(-tsShift5+tsWidth,-tsShift5+tsWidth*2,n5);
 
-  float ch3=corrCharge[3]/i3;
-  float ch4=(i3*corrCharge[4]-n3*corrCharge[3])/(i3*i4);
-  float ch5=(n3*n4*corrCharge[3]-i4*nn3*corrCharge[3]-i3*n4*corrCharge[4]+i3*i4*corrCharge[5])/(i3*i4*i5);
+  float ch3=0;
+  float ch4=0;
+  float ch5=0;
 
-  if (ch3<negThresh[0] && fNegStrat==HcalDeterministicFit::MoveCharge) {
-    ch3=negThresh[0];
-    ch4=corrCharge[4]/i4;
-    ch5=(i4*corrCharge[5]-n4*corrCharge[4])/(i4*i5);
-  }
+  if(i3==0 || i4==0 || i5==0) {
 
-  if (ch5<negThresh[0] && fNegStrat==HcalDeterministicFit::MoveCharge) {
-    ch4=ch4+(ch5-negThresh[0]);
-    ch5=negThresh[0];
-  }
+    ch3=0;
+    ch4=0;
+    ch5=0;
```

Comment:
Lines 129-131 are redundant -- these variables are already 0. Delete them and change the "else" to:
`if (i3 != 0 && i4 != 0 && i5 != 0) {`